src/Combinatorics/EnumerativeCombinatorics/partitions.jl

```diff
@@ -185,91 +221,93 @@ julia> collect(partitions(Int8(4))) # using less memory
  Int8[1, 1, 1, 1]
 ```
 """
-function partitions(n::T) where T <: IntegerUnion
+function partitions(n::IntegerUnion)
```

Comment:
Off-topic, but: I just noticed a function `partitions(n::Int, k::Int=n, m::Int=-1)` in `experimental/IntersectionTheory/src/Misc.jl` which of course clashes with the code here.

Which makes me wonder whether that function has any business sitting in `experimental/IntersectionTheory/src/Misc.jl`, and the whole file has a bunch of "interesting" content... huhrm